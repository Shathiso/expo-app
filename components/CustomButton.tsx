import { Text, TouchableOpacity, StyleSheet, View } from 'react-native'
import React from 'react'

import { FontAwesome } from '@expo/vector-icons';

interface props{
    title:string
    handlePress: () => void,
    isLoading: boolean,
    type: string
}
const CustomButton = ({title, isLoading, type, handlePress}:props) => {
  return (
    <TouchableOpacity
        onPress={handlePress}
        activeOpacity={0.7}
        style={ isLoading ? styles[`${type}Loading`] : styles[type]}
        disabled={isLoading}
    >
        <View style={styles.buttonTextWrapper} >
       <Text style={styles.text} >{title}</Text>
       {(type == 'signout') && <FontAwesome name="sign-out" size={24} color="#ad2524" />}
       </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
    text:{
        fontFamily:'Poppins-SemiBold',
        textAlign:'center',
        marginRight:5,
        marginTop:3.5,
    },
    buttonTextWrapper:{
        display:'flex',
        flexDirection:'row',
        justifyContent:'center',
    },
    primary:{
        backgroundColor: '#faa21b',
        paddingTop:15,
        paddingBottom:15,
        borderRadius: 12,
        textAlign:'center',
        shadowColor: '#000',
        shadowOffset: {
          width: 0,
          height: 2,
        },
        shadowOpacity: 0.40,
        shadowRadius: 4,
        elevation: 4
    },

    primarySingle:{
        backgroundColor: '#faa21b',
        paddingTop:15,
        paddingBottom:15,
        borderRadius: 12,
        textAlign:'center',
        marginTop:30
    },

    primaryLoading:{
        opacity:0.5,
        backgroundColor: '#faa21b',
        minHeight: 52,
        borderRadius: 12,
        marginTop:30
    },
    signout:{
        marginTop:20,
        marginRight:10,
        borderRadius: 8,
        width: 100
    }

});

export default CustomButton