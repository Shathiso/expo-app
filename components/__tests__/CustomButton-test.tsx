import * as React from 'react';
import { render, screen, fireEvent, userEvent } from '@testing-library/react-native';
import CustomButton from '../CustomButton';

jest.useFakeTimers();

test(`Custom Button functions correctly`, async () => {

  const user = userEvent.setup();
  const buttonPressHandler = jest.fn();

  const btnIsLoading = false;

  render(<CustomButton  title="Test" isLoading={btnIsLoading} type="primary" handlePress={buttonPressHandler} />);
  const screenBtn = screen.getByText('Test');
  await user.press(screenBtn);

  expect(buttonPressHandler).toHaveBeenCalled();

  const tree = render(<CustomButton  title="Test" isLoading={btnIsLoading} type="primary" handlePress={buttonPressHandler} />).toJSON();
  expect(tree).toMatchSnapshot();

});
