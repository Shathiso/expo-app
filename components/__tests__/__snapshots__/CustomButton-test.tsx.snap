// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Button functions correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#faa21b",
      "borderRadius": 12,
      "elevation": 4,
      "opacity": 1,
      "paddingBottom": 15,
      "paddingTop": 15,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.4,
      "shadowRadius": 4,
      "textAlign": "center",
    }
  }
>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "Poppins-SemiBold",
          "marginRight": 5,
          "marginTop": 3.5,
          "textAlign": "center",
        }
      }
    >
      Test
    </Text>
  </View>
</View>
`;
